File: Web/classes/Animal.js

export class Animal{
    //---------------------------------------
    // CONSTRUCTOR
    //---------------------------------------
    /*
    especie: Txt
    estado: Txt
    descripcion: Txt
    */
    //---------------------------------------
    constructor(especie,descripcion,caracteristicas,habitat,comida,estado){
        this._especie = especie;
        this._descripcion = descripcion;
        this._caracteristicas = caracteristicas;
        this._habitat = habitat;
        this._comida = comida;
        this._estado = estado;
    }
    //---------------------------------------
    // SETTERS y GETTERS
    //---------------------------------------
    set_especie(especie){
        this._especie=especie;
    }
    get_especie(){
        return this._especie;
    }
    //--------
    set_descripcion(descripcion){
        this._descripcion=descripcion;
    }
    get_descripcion(){
        return this._descripcion;
    }
    //--------
    set_caracteristicas(caracteristicas){
        this._caracteristicas=caracteristicas;
    }
    get_caracteristicas(){
        return this._caracteristicas;
    }
    //--------
    set_habitat(habitat){
        this._habitat=habitat;
    }
    get_habitat(){
        return this._habitat;
    }
    //--------
    set_comida(comida){
        this._comida=comida;
    }
    get_comida(){
        return this._comida;
    }
    //--------
    set_estado(estado){
        this._estado=estado;
    }
    get_estado(){
        return this._estado;
    }
    //--------
    //---------------------------------------
    // METODOS
    //---------------------------------------
    get_info(especie){
        fetch("http://localhost:5000/get_animal_info", {
            method: "POST",
            body: JSON.stringify({'especie':especie}),
            headers:{
                "Content-Type": "application/json",
            }
         })
         .then(info =>{
            // Code
            info.json().then(data =>{
                this.actualizar(data)
            })
         })
         .catch(error =>{
            // catch error
            console.log("Error: "+error)
         });
        //Fin
        console.log("get_info() hecho")
    }

    actualizar(data){
        this.set_especie(data["_especie"])
        this.set_descripcion(data["_descripcion"])
        this.set_caracteristicas(data["_caracteristicas"])
        this.set_habitat(data["_habitat"])
        this.set_comida(data["_comida"])
        this.set_estado(data["_estado"])
    }
}

File: Servidor/API/Animal.py
import os
import csv

class Animal:
  def __init__(self,especie, descripcion, caracteristicas, habitat, comida, estado):
    self._especie = especie
    self._descripcion = descripcion
    self._caracteristicas = caracteristicas
    self._habitat = habitat
    self._comida = comida
    self._estado = estado
    return
  
  def __init__(self):
    self._especie = None
    self._descripcion = None
    self._caracteristicas = None
    self._habitat = None
    self._comida = None
    self._estado = None
    return
#---------------------------------------------------
# GETTER
#---------------------------------------------------
  def get_especie(self):
    return self._especie
  def get_descripcion(self):
    return self._descripcion
  def get_caracteristicas(self):
    return self._caracteristicas
  def get_habitat(self):
    return self._habitat
  def get_comida(self):
    return self._comida
  def get_estado(self):
    return self._estado
    #---------------------------------------------------
    # SETTER
    #---------------------------------------------------
  def set_especie(self,especie):
    self._especie=especie
  def set_descripcion(self,descripcion):
    self._descripcion=descripcion
  def set_caracteristicas(self,caracteristicas):
    self._caracteristicas=caracteristicas
  def set_habitat(self,habitat):
    self._habitat=habitat
  def set_comida(self,comida):
    self._comida=comida
  def set_estado(self,estado):
      self._estado=estado
    #---------------------------------------------------
    # METODOS
    #---------------------------------------------------

      #---------------------------------------------------
      # especie:Txt --> get_info() --> animal:Animal
      # Descripción: Lee un csv y de sus datos devuelve la información correspondiente al animal
      # que se ha especificado con anterioridad
      #---------------------------------------------------
  def get_info(self, especie):
    datos = self.get_csv("../bbdd/animales.csv")
    
    dicres={}
    dicindex=[]
    index=None
    for item in datos:
      for i in item:
        if(item[i]==especie):
          index = i[-1]
          
    for item in datos:
      for i in item:
        if(i[-1]==index):
          dicindex.append(i)
          
    for item in datos:
      for index in dicindex:
        dicres[index]=item[index]
    
    self.set_parametros(dicres,index)
    return
      
  def get_csv(self,filename):
    dicres={}
    with open(filename, newline='', encoding="utf-8-sig") as csvfile:
      spamreader = csv.DictReader(csvfile, delimiter=';', quotechar='|')
      
      lista=[]
      index=1
      for row in spamreader:
        for i in row:
          #dicres[i+str(index)]=row.get(i)
          dicres[i+str(index)]=row.get(i)
        index+=1
      lista.append(dicres)
      
    return lista
  
  def set_parametros(self,datos,index):
    for item in datos:
      campo="_"+(item[:-1]).lower()
      if(campo=="_especie"):
        self.set_especie(datos[item])
      if(campo=="_descripcion"):
        self.set_descripcion(datos[item])
      if(campo=="_caracteristicas"):
        self.set_caracteristicas(datos[item])
      if(campo=="_habitat"):
        self.set_habitat(datos[item])
      if(campo=="_comida"):
        self.set_comida(datos[item])
      if(campo=="_estado"):
        self.set_estado(datos[item])
      else:
        self.set_estado("Desconocido")

    return